server/src/main/java/tw/waterball/survive/io/server/nio/NioServer.java

```diff
@@ -0,0 +1,98 @@
+package tw.waterball.survive.io.server.nio;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.nio.ByteBuffer;
+import java.nio.channels.*;
+
+import java.util.Iterator;
+
+/**
+ * @author swshawnwu@gmail.com(ShawnWu)
+ */
+public class NioServer {
+
+    private ServerSocketChannel socketChannel = ServerSocketChannel.open();
+    private Selector selector = Selector.open();
+    private SelectionKey selection;
+    private ByteBuffer buffer;
+    private boolean serverStatus;
+
+    private static NioServer nioServer;
+
+    public static NioServer create(int port) throws IOException {
+        if (nioServer == null)
+            nioServer = new NioServer(port);
+        return nioServer;
+    }
+
+    private NioServer(int port) throws IOException {
+        socketChannel.configureBlocking(false);
+        socketChannel.socket().bind(new InetSocketAddress(port));
+        socketChannel.register(selector, SelectionKey.OP_ACCEPT);
+    }
+
+    public void start() throws IOException {
+        serverStatus = true;
+        while (serverStatus) {
+            int eventSize = selector.select();
+            if (eventSize != 0) {
+                Iterator<SelectionKey> keyIterator = selector.selectedKeys().iterator();
+                while (keyIterator.hasNext()) {
+                    selection = keyIterator.next();
+                    keyIterator.remove();
```

Comment:
I'm not sure, could we use **for-each** here?

Reply:
Nice suggestion, I'll adopt it.

Reply:
But notice that the 43 line has `keyIterator.remove();` which might restrict the for-each loop adoption